notebooks/MAST/JWST/JWST_EngineeringDatabase_Retreival/JWST_EDB_Retrieval.ipynb

```diff
@@ -0,0 +1,560 @@
+    "\n",
+    "## Define Mnemonic Parameters\n",
+    "\n",
+    "Next, define the parameters of each mnemonic of interest. The parameters are:\n",
+    "* The mnenomic name\n",
+    "* Start time\n",
+    "* End time\n",
+    "\n",
+    "The start and end times are in UTC and have a \"compact\" ISO-8601 formatting: `yyyymmddThhmmss`, where the **T** is a literal character. The definitions can be stored in multiple ways: here they will be stored in a python dictionary, which could be stored in an external `.yaml` file. In the companion script they are stored in an external `.csv` file."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Since the mnemonics of interest are a tuple, the start/end times are the same: from 0:00 to 06:00 on 2022 June 01. Define these times first, followed by the full parameter dictionary."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "times = { \n",
+    "         't_start': '20220601T000000',\n",
+    "         't_end':   '20220601T060000'\n",
+    "        }\n",
+    "mnemonics = {\n",
+    "            'SA_ZADUCMDX': times,\n",
+    "            'SA_ZADUCMDY': times\n",
+    "           }\n",
+    "for m,v in mnemonics.items():\n",
+    "    print(m, v)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Construct File Names\n",
+    "\n",
+    "The key to fetching data from the web service is to construct file names to contain the data for each mnemonic. The web service will parse the file names to determine how to query the engineering database and retrieve the timeseries of interest.\n",
+    "\n",
+    "The file names have the form: \n",
+    "\n",
+    "    `<mnemonic_name>-<t_start>-<t_end>.csv`\n",
+    "    \n",
+    "Use a dictionary comprehension to construct a list of file names; these will be passed to the webservice calling function."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "fnames = ['-'.join([m, v['t_start'], v['t_end']]) + '.csv' for m,v in mnemonics.items()]\n",
+    "print(fnames)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Call the Webservice"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Set the (optional) output folder name prior to the webservice call. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Sub-directory where the data files will be written:\n",
+    "subdir = 'edb_test'"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now call the EDB web service. The files containing data will be written to your local storage, in the specified subdirectory. \n",
```

Comment:
I got this error, but it's probably because I am not on the authorized list to retrieve EDB data right now:

```
Downloading File: mast:jwstedb/SA_ZADUCMDX-20220601T000000-20220601T060000.csv
  To: edb_test/SA_ZADUCMDX-20220601T000000-20220601T060000.csv
  ***Error downloading file***
Downloading File: mast:jwstedb/SA_ZADUCMDY-20220601T000000-20220601T060000.csv
  To: edb_test/SA_ZADUCMDY-20220601T000000-20220601T060000.csv
  ***Error downloading file***
```

Reply:
Because of that, I can't actually run the rest of the notebook, so I will just comment on it without doing so past this point.